android/sdl_android/src/main/java/com/smartdevicelink/encoder/VirtualDisplayEncoder.java

```diff
@@ -258,6 +265,47 @@ public void shutDown() {
         }
     }
 
+    public void shutDown(boolean withPendingRestart) {
+        if (!initPassed) {
+            DebugTool.logError(TAG, "VirtualDisplayEncoder was not properly initialized with the init() method.");
+            return;
+        }
+        try {
+            // cleanup GLES stuff
+            if (mCaptureThread != null) {
+                mCaptureThread.stopAsync();
+                try {
+                    mCaptureThread.join();
+                } catch(InterruptedException e) {
+
+                }
+                mCaptureThread = null;
+            }
+            if (encoderThread != null) {
+                encoderThread.interrupt();
+                encoderThread = null;
+            }
+
+            if (mVideoEncoder != null) {
+                mVideoEncoder.stop();
+                mVideoEncoder.release();
+                mVideoEncoder = null;
+            }
+
+            if (virtualDisplay != null && !withPendingRestart) {
```

Comment:
@kostyaBoss As the VideoStreamService will be stopped and restarted, we can still dispose of the virtualDisplay in this case as it will be recreated when the VideoStreamService starts again.

Since this is the case we do not need a new shutDown method and can continue to use the old one

Reply:
@RHenigan Thanks for your reply. I still think we should not dispose the`virturalDisplay` because it leads to disposal of the `view` (overridden by the developer) bonded to `virtualDisplay` and as a result, leads to **loss of the state** each time when new capability will come (because the view is reflectively re-instantiated and bounded to the `virtualDisplay` on the system level). 

Indeed, according to the proposal, we should restart the **service**, but as for me, it doesn't seem like expected behaviour to have the state lost but not kept after all the things have been adjusted. 

For e.x if the developer will override the view with the WebView component inside -> the current web page on the new capability would be lost and the user will be forced to renavigate there

Please, let me know the next steps required assuming posted examples and arguments

Reply:
Hello @kostyaBoss after further discussion with the team it would fall on the developer to save the state of their presentation class that is extending the SdlRemoteDisplay. the library would then still dispose of the virtualDisplay and when the presentation is reinstantiated the developer should have the saved state available.

If the library is to handle the state of the stream there would need to be a change to the proposal